pootle/i18n/dates.py

```diff
@@ -35,14 +37,18 @@ def locale_code(self):
         return system_locale.getlocale()[0]
 
     def format_timesince(self, timestamp, locale=None):
-        return format_timedelta(
-            datetime.now()
-            - datetime.fromtimestamp(
-                timestamp),
-            locale=(
-                locale
-                or self.locale_code
-                or self.default_locale))
+        timedelta = datetime.now() - datetime.fromtimestamp(timestamp)
+        locale = (locale
+                  or self.locale_code
+                  or self.default_locale)
+        try:
+            return format_timedelta(timedelta, locale=locale)
+        except UnknownLocaleError:
+            locale = settings.LANGUAGE_CODE.split("-")[0]
+            try:
+                return format_timedelta(timedelta, locale=locale)
+            except ValueError:
+                return format_timedelta(timedelta, locale='en')
```

Comment:
This needs a test.

This seems like it's following the same pattern from the bits I coded.  Can you not refactor that out into a function that devolves to a workable locale.